app/Http/Controllers/AjaxController.php

```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
         }
         return response()->json(array("message" => "generic error message"), 500);
     }
+
+    /*
+     * AJAX Request to handle accepting/declining a friend request
+     *
+     */
+    public function processFriendRequest(Request $req) {
+        if ($req->json() && isset ($req->username)) {
+            $friendship = Friend::whereIn('username1', [Auth::user()->username, $req->username])
+                ->whereIn('username2', [Auth::user()->username, $req->username])
+                ->where('status_code', 'pending')
+                ->get();
```

Comment:
The query is wrong. You should be getting the rows where username2 is the Auth::user() and status_code is pending [and action_username is not username2]

The one in square bracket is optional